tests/MongoDB.Driver.Core.Tests/Core/Helpers/MockClusterableServerFactory.cs

```diff
@@ -43,16 +39,23 @@ public MockClusterableServerFactory(IEventSubscriber eventSubscriber = null)
         public IClusterableServer CreateServer(ClusterId clusterId, IClusterClock clusterClock, EndPoint endPoint)
         {
             ServerTuple result;
-            if (!_servers.TryGetValue(endPoint, out result))
+            if (!_servers.TryGetValue(endPoint, out result) || result.HasBeenRemoved)
```

Comment:
if the server has been removed, we possible will need to recalculate his description